tests/Tests/600__ai_apps/600__anaconda_commercial_edition/600__anaconda_commercial_edition.robot

```diff
@@ -75,4 +99,60 @@ Get Connect Button Status
   ${button_status}=  Get Element Attribute    xpath://*/footer/*[.='Connect']    aria-disabled
   [Return]   ${button_status}
 
+Get Anaconda Validator Logs
+  Go To  ${OCP_CONSOLE_URL}
+  Login To Openshift    ${OCP_ADMIN_USER.USERNAME}    ${OCP_ADMIN_USER.PASSWORD}    ${OCP_ADMIN_USER.AUTH_TYPE}
+  Maybe Skip Tour
+  Navigate To Page    Workloads    Pods
+  Search Last Item Instance By Title in OpenShift Table  search_term=anaconda-ce-periodic-validator-job-custom-run  namespace=redhat-ods-applications
+  Click Link    xpath://tr[@data-id='0-0']/td[@id='name']/*/a
+  Click Link    Logs
+  Sleep  2
+  Capture Page Screenshot  logs_page.png
+  ${logs_text}=  Get Text    xpath://div[@class='log-window__lines']
+  ${log_rows}=  Text To List  ${logs_text}
+  [Return]  ${log_rows[0]}
+
+Check Anaconda CE Image Build Status
+  [Arguments]  ${target_status}
+  Navigate To Page    Builds    Builds
+  Search Last Item Instance By Title in OpenShift Table  search_term=minimal-notebook-anaconda  namespace=redhat-ods-applications
+  ${ace_build_status}=  Get Text    xpath://tr[@data-id='0-0']/td/*/span[@data-test='status-text']
+  Run Keyword If    "${ace_build_status}" == "Failed"
+  ...    Fail  the Anaconda image build has failed
+  ...  ELSE
+  ...    Should Be Equal    ${ace_build_status}    ${target_status}
+
+Spawn and Check JupyterLab Using Anaconda CE Image
+  Open Browser  ${ODH_DASHBOARD_URL}  browser=${BROWSER.NAME}  options=${BROWSER.OPTIONS}
+  Login To RHODS Dashboard  ${TEST_USER.USERNAME}  ${TEST_USER.PASSWORD}  ${TEST_USER.AUTH_TYPE}
+  Menu.Navigate To Page    Applications    Enabled
+  Launch JupyterHub From RHODS Dashboard Dropdown
+  Login To Jupyterhub  ${TEST_USER.USERNAME}  ${TEST_USER.PASSWORD}  ${TEST_USER.AUTH_TYPE}
+  ${authorization_required} =  Is Service Account Authorization Required
+  Run Keyword If  ${authorization_required}  Authorize jupyterhub service account
+  Fix Spawner Status
+  Wait Until Page Contains Element  xpath://span[@id='jupyterhub-logo']
+  Wait Until Page Contains Element  xpath://input[@name='Anaconda Commercial Edition']
+  Wait Until Element Is Enabled    xpath://input[@name='Anaconda Commercial Edition']   timeout=600
+  Spawn Notebook With Arguments  image=s2i-minimal-notebook-anaconda
+  Wait for JupyterLab Splash Screen  timeout=60
```

Comment:
This can be abstracted in a keyword, but it should be a general keyword rather than specific for anaconda

Reply:
Agree. I'm going to extract a generic keyword from here and put it in the JupyterHubSpawner.robot (i.e., Spawn Notebook From Dashboard With Arguments). The remaining lines of code will be moved in the TC.